d2layouts/d2elklayout/layout.go

```diff
@@ -525,30 +652,40 @@ func deleteBends(g *d2graph.Graph) {
 			var corner *geo.Point
 			var end *geo.Point
 
+			var columnIndex *int
 			if isSource {
 				start = e.Route[0]
 				corner = e.Route[1]
 				end = e.Route[2]
 				endpoint = e.Src
+				columnIndex = e.SrcTableColumnIndex
 			} else {
 				start = e.Route[len(e.Route)-1]
 				corner = e.Route[len(e.Route)-2]
 				end = e.Route[len(e.Route)-3]
 				endpoint = e.Dst
+				columnIndex = e.DstTableColumnIndex
 			}
 
 			isHorizontal := math.Ceil(start.Y) == math.Ceil(corner.Y)
 			dx, dy := endpoint.GetModifierElementAdjustments()
 
 			// Make sure it's still attached
-			if isHorizontal {
+			switch {
+			case columnIndex != nil:
+				maxRowOffset := endpoint.Height / float64(len(endpoint.SQLTable.Columns)+1) / 3
+				rowCenter := endpoint.TopLeft.Y + maxRowOffset*float64(*columnIndex+1)*2 + maxRowOffset
+				if math.Abs(end.Y-rowCenter) > maxRowOffset {
+					continue
+				}
```

Comment:
```suggestion
				rowHeight := endpoint.Height / float64(len(endpoint.SQLTable.Columns)+1)
				rowCenter := endpoint.TopLeft.Y + rowHeight*float64(*columnIndex+1) + rowHeight/2

				// for row connections new Y coordinate should be within 1/3 row height from the row center
				if math.Abs(end.Y-rowCenter) > rowHeight /3 {
					continue
				}
```

it looks like there might be a bug in the rowCenter calculation, this suggestion looks clearer to me

Reply:
`maxRowOffset` was originally equal to half row height. It was too much so I changed it to 1/3 but the rowCenter calculation remained as it was...

Your suggestion entirely fixes what I was fighting with:

<img width="467" alt="Screenshot 2023-10-25 at 08 37 22" src="https://github.com/terrastruct/d2/assets/13437951/0ae6665c-a9ee-43dc-b6ea-a038e931c4cb">

Thanks!